### ISSUE TITLE
Framework not linked from dependent project

### ISSUE THREAD

**Issue by abbeycode** [CocoaPods/CocoaPods#4381]
I'm investigating an issue a user reported with my library (https://github.com/abbeycode/UnzipKit/issues/21). In the demo project he submitted, he has a pretty simple Podfile:

```
source 'https://github.com/CocoaPods/Specs.git'

platform :ios, "8.0"

link_with 'UsingUnzipKit'

inhibit_all_warnings!
use_frameworks!

pod 'UnzipKit'
```

After a `pod install`, though, it looks like the only framework showing up in the source project is `Pods.framework`, even though the Pods project does produce an `UnzipKit.framework`. The Pods framework doesn't contain any references to UnzipKit in its umbrella header. Is this correct behavior? It looks like either the Pods.framework should include all of UnzipKit, or CocoaPods should link the project against `UnzipKit.framework` directly. I'd appreciate any assistance you can offer.

<img width="761" alt="Xcode Screenshot" src="https://cloud.githubusercontent.com/assets/1349578/10459927/fcb84008-719f-11e5-809f-c9cb00772834.png">

**Comment by segiddins** [comment]
This is intentional, the XCConfig files will include the necessary linker flags for `UnzipKit.framework` and the actual wrapper will be copied into the app via the `Embed Pods Frameworks` script build phase.

**Comment by abbeycode** [comment]
Oh ok, that makes sense. The target with build issues here is a Framework target, and when adding either `@import UnzipKit;` or `#import <UnzipKit/UnzipKit.h>` to the umbrella header, we're getting errors for 4 different headers in UnzipKit:

> Include of non-modular header inside framework module 'UnzipKit.<HeaderName>'

That's happening for each header at the point where it calls `#include "zlib.h"`, which is part of `libz.dylib`. I verified that the UnzipKit Pods target includes a `-l"z"` linker flag, and adding same to the target project didn't help it to compile.

I also tried turning on `CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES` in the Pods project and the target project, and it made no difference.
